database/models.py

```diff
@@ -25,6 +25,7 @@ class Experiment(Base):
 
     name = Column(String, nullable=False, primary_key=True)
     time_created = Column(DateTime(), server_default=sqlalchemy.func.now())
+    time_ended = Column(DateTime(), nullable=True)
```

Comment:
Given how often we shut things down early I think this is going to be pretty fragile.

Reply:
We shouldn't. And there should be a field to indicate experiment finished, otherwise there isn't one. I feel service is working fine, what is the issue ? Also, this just means that incomplete experiment won't be used for merging (like if you shut it off), which also makes sense.

Reply:
Large scale preemptible experiments seem to be broken and so they need to be shutdown manually or else they never finish.
Obviously that issue should be fixed and I think we should also add some sanity checking to measuring so that we dont retry forever.